packages/x-components/src/components/base-event-button.vue

```diff
@@ -26,6 +27,14 @@
       events: {
         type: Object as PropType<Partial<XEventsTypes>>,
         required: true
+      },
+      /**
+       * The origin property for the request on each query preview.
+       *
+       * @public
+       */
+      queryFeature: {
```

Comment:
I think would be better to not limit this to the queryFeature. I'd something like:

```ts
      metadata: {
        type: Object as PropType<Omit<WireMetadata, 'moduleName'>>
      }
...
      function emitEvents() {
        Object.entries(props.events).forEach(([event, payload]) =>
          $x.emit(event as XEvent, payload, { target: rootRef.value, ...props.metadata })
        );
      }
```
I'm not sure if we could have issues with the wiring types. If we do, probably we could even relax them here.

Reply:
You're right!! 👍🏽